tensorflow/compiler/mlir/lite/tests/legalize-tf.mlir

```diff
@@ -127,6 +127,28 @@ func.func @avgPool2D(%arg0: tensor<1x6x6x16xf32>) -> tensor<1x1x1x16xf32> {
 // CHECK:  %3 = "tf.AvgPool"(%arg0)
 }
 
+func.func @avgPool3D(%arg0: tensor<1x6x6x6x16xf32>) -> tensor<1x1x1x1x16xf32> {
+  // OK
+  %0 = "tf.AvgPool3D"(%arg0) {T = "tfdtype$DT_FLOAT", data_format = "NDHWC", ksize = [1, 2, 3, 6, 1], padding = "VALID", strides = [1, 2, 3, 1, 1]} : (tensor<1x6x6x6x16xf32>) -> tensor<1x1x1x1x16xf32>
+  // Unsupported data format
+  %1 = "tf.AvgPool3D"(%arg0) {T = "tfdtype$DT_FLOAT", data_format = "NCDHW", ksize = [1, 2, 3, 6, 1], padding = "VALID", strides = [1, 2, 3, 1, 1]} : (tensor<1x6x6x6x16xf32>) -> tensor<1x1x1x1x16xf32>
+  // Unsupported ksize
+  %2 = "tf.AvgPool3D"(%arg0) {T = "tfdtype$DT_FLOAT", data_format = "NDHWC", ksize = [3, 2, 3, 6, 1], padding = "VALID", strides = [1, 2, 3, 1, 1]} : (tensor<1x6x6x6x16xf32>) -> tensor<1x1x1x1x16xf32>
+  // Unsupported strides
+  %3 = "tf.AvgPool3D"(%arg0) {T = "tfdtype$DT_FLOAT", data_format = "NDHWC", ksize = [1, 2, 3, 6, 1], padding = "VALID", strides = [1, 2, 3, 1, 3]} : (tensor<1x6x6x6x16xf32>) -> tensor<1x1x1x1x16xf32>
+
+  %5 = arith.addf %0, %1 : tensor<1x1x1x1x16xf32>
```

Comment:
I'm assuming you are doing this just to avoid dead code elimination? If so, how about using tf.AddN with a comment.

Reply:
From my understanding it was done to make it easier to have a single return tensor without having to mess with shapes. It's done in a similar way for 2D average and max pooling tests.